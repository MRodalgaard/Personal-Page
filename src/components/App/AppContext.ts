import { createContext } from 'react';
import { AppBackground } from '../../util/constants';
import { AppColors } from '../../util/theme';

export interface IAppState {
  color: AppColors;
  background: AppBackground;
  toggleBackground?: () => void;
  toggleColor?: () => void;
}

export const initialState: IAppState = {
  background: AppBackground.greyscale,
  color: AppColors.primary,
};

const AppContext = createContext(initialState);

export default AppContext;
