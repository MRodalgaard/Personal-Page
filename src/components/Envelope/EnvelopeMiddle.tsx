import styled from 'styled-components';

const Container = styled.header`
  align-self: center;
`;

const Heading = styled.h2`
  font-size: 1.5rem;
  line-height: 1.5rem;
`;

export default function EnvelopeMiddle({ children }: { children?: React.ReactNode }) {
  return (
    <Container>
      <Heading>{children}</Heading>
    </Container>
  );
}
