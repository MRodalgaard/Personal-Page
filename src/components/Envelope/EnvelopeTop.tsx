import styled from 'styled-components';

export const EnvelopeTop = styled.div`
  display: flex;
  place-items: start;
  gap: 4px;

  *:last-child {
    margin-left: auto;
  }
`;
