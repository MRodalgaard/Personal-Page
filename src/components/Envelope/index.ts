export { Envelope } from './Envelope';
export { EnvelopeMiddle } from './EnvelopeMiddle';
export { EnvelopeTop } from './EnvelopeTop';
