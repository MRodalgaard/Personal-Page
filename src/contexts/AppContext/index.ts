export { AppContextProvider } from './AppContextProvider';
export { useAppContext } from './useAppContext';
