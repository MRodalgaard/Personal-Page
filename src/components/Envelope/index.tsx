import { faFacebookF } from '@fortawesome/free-brands-svg-icons/faFacebookF';
import { faGithub } from '@fortawesome/free-brands-svg-icons/faGithub';
import { faLinkedin } from '@fortawesome/free-brands-svg-icons/faLinkedin';
import { faTwitter } from '@fortawesome/free-brands-svg-icons/faTwitter';
import * as React from 'react';
import { useContext } from 'react';
import styled from 'styled-components';
import { LogEvent } from '../../util/analytics';
import { AppColors } from '../../util/theme';
import AppContext from '../App/AppContext';
import MailLink from '../MailLink';
import { Paper } from '../Shared/Paper';
import SocialLink from '../SocialLink';

const EnvelopeTop = styled.div`
  border-bottom: 60px solid ${AppColors.grey};
  border-left: 50px solid transparent;
  border-right: 50px solid transparent;
`;

const EnvelopeBody = styled(Paper)`
  height: 200px;
  padding: 5px;
  font-size: 20px;
`;

const EnvelopeText = styled.p`
  width: 50%;
  height: 60px;
  font-weight: bold;
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
`;

const Envelope = () => {
  const { color } = useContext(AppContext);

  return (
    <>
      <EnvelopeTop />
      <EnvelopeBody>
        <SocialLink
          href="https://facebook.com/mrodalgaard"
          icon={faFacebookF}
          color={color}
          logEvent={LogEvent.FacebookLink}
        />
        <SocialLink
          href="https://linkedin.com/in/mrodalgaard"
          icon={faLinkedin}
          color={color}
          logEvent={LogEvent.LinkedInLink}
        />
        <SocialLink
          href="https://github.com/mrodalgaard"
          icon={faGithub}
          color={color}
          logEvent={LogEvent.GithubLink}
        />
        <SocialLink
          href="https://twitter.com/mrodalgaard"
          icon={faTwitter}
          color={color}
          logEvent={LogEvent.TwitterLink}
        />
        <MailLink href="mailto:mrodalgaard@gmail.com" color={color} />
        <EnvelopeText>
          Martin Rodalgaard
          <br />
          Aarhus
          <br />
          Denmark
        </EnvelopeText>
      </EnvelopeBody>
    </>
  );
};

export default Envelope;
