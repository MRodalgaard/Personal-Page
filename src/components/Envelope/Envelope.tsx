import { Draggable, DraggableIcon } from 'components/Draggable';
import { Paper } from 'components/Paper';
import styled from 'styled-components';

const EnvelopeFlap = styled.div`
  border-bottom: 60px solid ${({ theme }) => theme.colors.primary};
  border-left: 50px solid transparent;
  border-right: 50px solid transparent;
`;

const EnvelopeBody = styled(Paper)`
  display: flex;
  flex-direction: column;
  height: ${({ theme }) => theme.envelopeHeight}px;
  padding: 5px;
`;

export const Envelope = ({ children }: { children?: React.ReactNode }) => {
  return (
    <Draggable>
      <EnvelopeFlap />
      <EnvelopeBody>
        {children}
        <DraggableIcon position="bottom-right" />
      </EnvelopeBody>
    </Draggable>
  );
};
