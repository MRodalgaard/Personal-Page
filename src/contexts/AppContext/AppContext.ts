import { AppTheme } from 'contexts/ThemeContext';
import { createContext } from 'react';

export interface IAppContext {
  colorized: boolean;
  theme: AppTheme;
  toggleColorized: () => void;
  toggleTheme: () => void;
}

export const defaultValue: IAppContext = {
  colorized: false,
  theme: 'light',
  toggleColorized: () => {},
  toggleTheme: () => {},
};

export const AppContext = createContext<IAppContext>(defaultValue);
