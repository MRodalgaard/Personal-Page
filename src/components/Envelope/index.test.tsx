import { render } from '@testing-library/react';
import React from 'react';
import Envelope from '.';
import AppContext, { initialState } from '../App/AppContext';

describe('Envelope', () => {
  it('renders', () => {
    const { container } = render(
      <AppContext.Provider value={initialState}>
        <Envelope />
      </AppContext.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
