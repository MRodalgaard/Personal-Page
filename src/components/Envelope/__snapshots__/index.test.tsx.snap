// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Envelope renders 1`] = `
.c0 {
  border-bottom: 60px solid #333333;
  border-left: 50px solid transparent;
  border-right: 50px solid transparent;
}

<div
  class="c0"
/>
`;
